### DISCUSSION TITLE
Not able to cancel the job from client application

### DISCUSSION THREAD

**Discussion by kibo-raghu** [quartznet/quartznet#953] [General]
I am using Quartz.net 3.0.7 and hosted it in docker using IHostedService. I have a website where users can schedule and cancel the jobs they scheduled. Scheduling job works fine but Scheduler.Interrup() does not do anything to interrupt the running job from website. How do I implement cancellation feature from the website? How do I get the already running scheduler object to tweak the things here?

**Comment by lahma** [comment]
There's an example about this [here](https://github.com/quartznet/quartznet/tree/master/src/Quartz.Examples/07_InterrupInProgressJobs).

**Comment by kibo-raghu** [comment]
In the example provided above the scheduler and interrupt are run in the same class and interrupt would have access to scheduler object. My requirement is different(I am using ADOJobStore here(SQL Server)):

1. I have a scheduler and corresponding jobs running on docker containers.
2. I have a website running on different server and can schedule a job because all it needs is a data entry to the database tables.
3. Similarly I wanted to interrupt a long running job. I will not have access to already existing scheduler object running on the another container. I tried creating another scheduler object with same configuration in the UI and tried to use scheduler.Interrupt(),but no luck.  How do I implement cancelling a job in this case?

I hope I made myself clear here.Let me know if you need any details.

**Reply by lahma** [reply]
In your case you have the same restriction like in a cluster, interrupt is not cluster aware so you would need to create your own flag to check whether the job should still continue to process.
